q2_moshpit/plugin_setup.py

```diff
@@ -1460,3 +1465,40 @@
     ReferenceDB[BuscoDB], BuscoDatabaseDirFmt
 )
 importlib.import_module('q2_moshpit.busco.types._transformer')
+
+plugin.methods.register_function(
+    function=q2_moshpit.eggnog.fetch_eggnog_hmmer_db,
+    inputs={},
+    parameters={
+        "taxon_id": Int % Range(1, None)
```

Comment:
When I try to run it with `1` I get an error:
```
Validating taxon ID:
Downloading taxonomy file...
Running external command line application(s). This may print messages to stdout and/or stderr.
The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.

Command: wget -O /var/folders/7f/7nw_x13n5q965rss_qz6061m0000gq/T/tmpkuvn0wmy/e5.taxid_info.tsv http://eggnog5.embl.de/download/eggnog_5.0/e5.taxid_info.tsv

--2024-06-20 13:08:45--  http://eggnog5.embl.de/download/eggnog_5.0/e5.taxid_info.tsv
Resolving eggnog5.embl.de (eggnog5.embl.de)... 194.94.44.170
Connecting to eggnog5.embl.de (eggnog5.embl.de)|194.94.44.170|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1429293 (1.4M) [application/octet-stream]
Saving to: ‘/var/folders/7f/7nw_x13n5q965rss_qz6061m0000gq/T/tmpkuvn0wmy/e5.taxid_info.tsv’

/var/folders/7f/7nw_x13n5q965rss_qz6061m0000gq/T/tmpkuvn0wmy/e5.taxid_info.tsv               100%[============================================================================================================================================================================================================================================>]   1.36M  1.17MB/s    in 1.2s

2024-06-20 13:08:47 (1.17 MB/s) - ‘/var/folders/7f/7nw_x13n5q965rss_qz6061m0000gq/T/tmpkuvn0wmy/e5.taxid_info.tsv’ saved [1429293/1429293]

Valid taxon ID.
Proceeding with HMMER database download and build...
Running external command line application(s). This may print messages to stdout and/or stderr.
The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.

Command: wget -O /var/folders/7f/7nw_x13n5q965rss_qz6061m0000gq/T/tmpxaz500ny/1_hmms.tar.gz http://eggnog5.embl.de/download/eggnog_5.0/per_tax_level/1/1_hmms.tar.gz

--2024-06-20 13:08:47--  http://eggnog5.embl.de/download/eggnog_5.0/per_tax_level/1/1_hmms.tar.gz
Resolving eggnog5.embl.de (eggnog5.embl.de)... 194.94.44.170
Connecting to eggnog5.embl.de (eggnog5.embl.de)|194.94.44.170|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 56 [application/octet-stream]
Saving to: ‘/var/folders/7f/7nw_x13n5q965rss_qz6061m0000gq/T/tmpxaz500ny/1_hmms.tar.gz’

/var/folders/7f/7nw_x13n5q965rss_qz6061m0000gq/T/tmpxaz500ny/1_hmms.tar.gz                   100%[============================================================================================================================================================================================================================================>]      56  --.-KB/s    in 0s

2024-06-20 13:08:47 (4.86 MB/s) - ‘/var/folders/7f/7nw_x13n5q965rss_qz6061m0000gq/T/tmpxaz500ny/1_hmms.tar.gz’ saved [56/56]

Decompressing...
Running external command line application(s). This may print messages to stdout and/or stderr.
The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.

Command: tar zxf 1_hmms.tar.gz

Merging hmm files...
Preparing HMM database...
Running external command line application(s). This may print messages to stdout and/or stderr.
The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.

Command: hmmpress /var/folders/7f/7nw_x13n5q965rss_qz6061m0000gq/T/q2-PressedProfileHmmsDirectoryFmt-0kxly8rr/1.hmm


Error: File format problem in trying to open HMM file /var/folders/7f/7nw_x13n5q965rss_qz6061m0000gq/T/q2-PressedProfileHmmsDirectoryFmt-0kxly8rr/1.hmm.
File exists, but appears to be empty?

Traceback (most recent call last):
  File "/Users/mziemski/miniconda3/envs/q2-metagenome-2024.5/lib/python3.9/site-packages/q2cli/commands.py", line 520, in __call__
    results = self._execute_action(
  File "/Users/mziemski/miniconda3/envs/q2-metagenome-2024.5/lib/python3.9/site-packages/q2cli/commands.py", line 581, in _execute_action
    results = action(**arguments)
  File "<decorator-gen-758>", line 2, in fetch_eggnog_hmmer_db
  File "/Users/mziemski/miniconda3/envs/q2-metagenome-2024.5/lib/python3.9/site-packages/qiime2/sdk/action.py", line 342, in bound_callable
    outputs = self._callable_executor_(
  File "/Users/mziemski/miniconda3/envs/q2-metagenome-2024.5/lib/python3.9/site-packages/qiime2/sdk/action.py", line 576, in _callable_executor_
    output_views = self._callable(**view_args)
  File "/Users/mziemski/miniconda3/envs/q2-metagenome-2024.5/lib/python3.9/site-packages/q2_moshpit/eggnog/_dbs.py", line 362, in fetch_eggnog_hmmer_db
    idmap, hmmer_db, pressed_hmmer_db = _download_and_build_hmm_db(taxon_id)
  File "/Users/mziemski/miniconda3/envs/q2-metagenome-2024.5/lib/python3.9/site-packages/q2_moshpit/eggnog/_utils.py", line 103, in _download_and_build_hmm_db
    run_command(cmd=["hmmpress", hmms_merged_p])
  File "/Users/mziemski/miniconda3/envs/q2-metagenome-2024.5/lib/python3.9/site-packages/q2_moshpit/_utils.py", line 37, in run_command
    subprocess.run(cmd, check=True, **kwargs)
  File "/Users/mziemski/miniconda3/envs/q2-metagenome-2024.5/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['hmmpress', '/var/folders/7f/7nw_x13n5q965rss_qz6061m0000gq/T/q2-PressedProfileHmmsDirectoryFmt-0kxly8rr/1.hmm']' returned non-zero exit status 1.

Plugin error from moshpit:

  Command '['hmmpress', '/var/folders/7f/7nw_x13n5q965rss_qz6061m0000gq/T/q2-PressedProfileHmmsDirectoryFmt-0kxly8rr/1.hmm']' returned non-zero exit status 1.

See above for debug info.
```
Does not happen with `2` (or any other I tried).

Reply:
@misialq waa... this is weird. 1 is the root of the taxonomic tree no so it should essentially contain every taxa. Instead `1_hmms.tar.gz` is empty. But it exists in the ftp server. I would just say we limit the range from 2 onwards.